operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java

```diff
@@ -100,19 +100,30 @@ public static Condition buildRebalanceCondition(String type) {
         if (resource.getMetadata().getGeneration() != null)    {
             status.setObservedGeneration(resource.getMetadata().getGeneration());
         }
+        // condition contains a timestamp so using a set does not help
         List<Condition> conditionList = new ArrayList<>();
         Condition condition = StatusUtils.buildCondition(type, conditionStatus, null);
         conditionList.add(condition);
 
         if (status.getConditions() != null) {
             status.getConditions().forEach(cond -> {
                 if ("UnknownFields".equals(cond.getReason()) || "DeprecatedFields".equals(cond.getReason())) {
-                    conditionList.add(cond);
+                    if (!conditionAlreadyPresent(conditionList, cond)) {
+                        conditionList.add(cond);
```

Comment:
just trying to understand, why we should not put an already existing condition. If the same condition is happening, we should have the same one but with a different timestamp to show its latest update.

Reply:
It would result in this: https://gist.github.com/im-konge/b26e068fce4c5c1395717ef7c4a807ab

Reply:
how does it work today without warnings then?

Reply:
Before #4291 there is nothing besides of Proposal status.

Reply:
this is part of a proposal status but we didn't have these warnings conditions and the condition was always just one, or? Now we are moving to have more conditions then? @tomncooper do I remember wrong?

Reply:
Correct. The issue is described in https://github.com/strimzi/strimzi-kafka-operator/issues/3637